common/src/stack/command/stack/commands/sync/firmware/imp_m7520.py

```diff
@@ -0,0 +1,58 @@
+import stack.commands
+from stack.switch.m7800 import SwitchMellanoxM7800
+
+class Implementation(stack.commands.Implementation):
+
+	def run(self, args):
+		switch = args[0]['host']
+		image_name = args[1]
+
+		switch_attrs = self.owner.getHostAttrDict(switch)
+
+		kwargs = {
+			'username': switch_attrs[switch].get('username'),
+			'password': switch_attrs[switch].get('password')
+		}
+
+		kwargs = {k:v for k, v in kwargs.items() if v is not None}
+
+		s = SwitchMellanoxM7800(switch, **kwargs)
+		s.connect()
+		show_images = s.show_images()
+		images_to_delete = show_images['Available images']
```

Comment:
No need for intermediate variables.

```
for image in s.show_images()['available']:
    s.image_delete(image)
```

Even better would be if `image_delete()` (perhaps renamed to `delete_images`) when passed nothing would just call show images itself and delete them all.